_docs/_latest/develop/dev/inventory.md

```diff
@@ -5,49 +5,170 @@ order: 102
+```
+
+Create XYZIterator to call the iter_xyz(), and cast the result for storage
+in Inventory.
+
+```
+google/cloud/forseti/services/inventory/base/resources.py
+
+    class ImageIterator(ResourceIterator):
+        def iter(self):
+            gcp = self.client
+            if (self.resource.enumerable() and
+                   self.resource.compute_api_enabled(gcp)):
+                for data in gcp.iter_images(
+                        projectid=self.resource['projectId']):
+                    yield FACTORIES['image'].create_new(data)
+```
+
+In order to do the casting, we need to create a resource class for xyz. This
+allows us to access the `id` and `type`. If the resource has no provided `id`,
+we must synthesize our own.  See other existing `key()`, to see how to create
+a synthetic key.
+
+```
+google/cloud/forseti/services/inventory/base/resources.py
+
+    class Image(Resource):
+        def key(self):
+            return self['id']
+
+        def type(self):
+            return 'image'
+```
+
+## Add the iterator to Inventory factories to make it available for crawling
+
+Create ResourceFactory for xyz, and link the XYZIterator to the parent resource.
+
+```
+google/cloud/forseti/services/inventory/base/resources.py
+
+    FACTORIES = {
+        'project': ResourceFactory({
+            'dependsOn': ['organization', 'folder'],
+            'cls': Project,
+            'contains': [
+                ImageIterator,
+                FirewallIterator,
+                NetworkIterator,
+                SubnetworkIterator,
+            ]}),
+       'image': ResourceFactory({
+            'dependsOn': ['project'],
+            'cls': Image,
+            'contains': [
+                ]}),
+    }
+```
+
+## Add the Inventory data to data model
+
+Assumption here is that this is a simple resource, and will go into the
+resource data model table. If you need to convert the Inventory data into
+more complicated data model, please email `discuss@forsetisecurity.org` for
+further assistance.
+
+Add xyz into `gcp_type_list`, and create `_convert_xyz()` to store the inventory
+data into data model. Connect xyz with the `_convert_xyz()` in the h`andlers`
+map in `_store_resource()`.
+
+```
+google/cloud/forseti/services/model/importer/importer.py
+
+    def run(self):
+        """Runs the import.
+
+        Raises:
+            NotImplementedError: If the importer encounters an unknown
+                inventory type.
+        """
+        gcp_type_list = [
+            'organization',
+            'folder',
+            'project',
+            'image',
+        ]
+```
+
+```
+google/cloud/forseti/services/model/importer/importer.py
+
+    def _convert_image(self, image):
+        """Convert a image to a database object.
+
+         Args:
+            image (object): Image to store.
+        """
+        data = image.get_data()
+        parent, full_res_name, type_name = self._full_resource_name(
+            image)
+        self.session.add(
```

Comment:
What happens when this changes? We probably shouldn't be copying code into the docs, rather refer to github or the API docs that are generated.

Reply:
If the code changes here, we will have to update this. My perspective is that for a code-walk, actual code example is really helpful to give people a concrete overview and understanding of what is happening.  And that's why there is link to the actual code for them to reference the current state of the code.